store/userinfo.go

```diff
@@ -47,32 +50,48 @@ type User struct {
 
 func UserInfo(email string) (userinfo *User, err error) {
 	ssourl := fmt.Sprintf("%s/keys/%s", authURL(), url.QueryEscape(email))
+	for attempt := retry.Start(defaultRetryStrategy, nil); attempt.Next(); {
+		var resp *http.Response
+		resp, err = httpClient.Get(ssourl)
+		if err != nil {
+			if shouldRetryError(attempt, err) {
+				continue
+			}
+			break
+		}
 
-	resp, err := httpClient.Get(ssourl)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+		if shouldRetryHttpResponse(attempt, resp) {
+			resp.Body.Close()
+			continue
+		}
 
-	switch resp.StatusCode {
-	case 200:
-		// good
-		break
-	case 404:
-		return nil, fmt.Errorf("cannot find user %q", email)
-	default:
-		return nil, respToError(resp, fmt.Sprintf("look up user %q", email))
-	}
+		defer resp.Body.Close()
+
+		switch resp.StatusCode {
+		case 200:
+			// good
+			break
```

Comment:
The break here (which was already here before) is unnecessary and slightly misleading. It doesn't do anything useful here, so it looks like it's breaking out of the `for`, but it's simply doing nothing.